Source/Fuse.Nodes/VisualBounds.uno

```diff
@@ -245,21 +245,79 @@ namespace Fuse
-			float3 E = Vector.TransformCoordinate(float3(box.Minimum.X, box.Minimum.Y, box.Maximum.Z), transform);
-			float3 F = Vector.TransformCoordinate(float3(box.Maximum.X, box.Minimum.Y, box.Maximum.Z), transform);
-			float3 G = Vector.TransformCoordinate(float3(box.Maximum.X, box.Maximum.Y, box.Maximum.Z), transform);
-			float3 H = Vector.TransformCoordinate(float3(box.Minimum.X, box.Maximum.Y, box.Maximum.Z), transform);
+			return BoxTransform(box, FastMatrix.FromFloat4x4(transform));
+		}
+
+		//uses the W paramete runlike Box.Transform (which may be a defect there)
+		public static Box BoxTransform(Box box, FastMatrix matrix)
+		{
+			float3 A = matrix.TransformVector(float3(box.Minimum.X, box.Minimum.Y, box.Minimum.Z));
+			float3 B = matrix.TransformVector(float3(box.Maximum.X, box.Minimum.Y, box.Minimum.Z));
+			float3 C = matrix.TransformVector(float3(box.Maximum.X, box.Maximum.Y, box.Minimum.Z));
+			float3 D = matrix.TransformVector(float3(box.Minimum.X, box.Maximum.Y, box.Minimum.Z));
+			float3 E = matrix.TransformVector(float3(box.Minimum.X, box.Minimum.Y, box.Maximum.Z));
+			float3 F = matrix.TransformVector(float3(box.Maximum.X, box.Minimum.Y, box.Maximum.Z));
+			float3 G = matrix.TransformVector(float3(box.Maximum.X, box.Maximum.Y, box.Maximum.Z));
+			float3 H = matrix.TransformVector(float3(box.Minimum.X, box.Maximum.Y, box.Maximum.Z));
+
+			float minX = A.X;
+			if (B.X < minX) minX = B.X;
+			if (C.X < minX) minX = C.X;
+			if (D.X < minX) minX = D.X;
+			if (E.X < minX) minX = E.X;
+			if (F.X < minX) minX = F.X;
+			if (G.X < minX) minX = G.X;
+			if (H.X < minX) minX = H.X;
+
+			float minY = A.Y;
+			if (B.Y < minY) minY = B.Y;
+			if (C.Y < minY) minY = C.Y;
+			if (D.Y < minY) minY = D.Y;
+			if (E.Y < minY) minY = E.Y;
+			if (F.Y < minY) minY = F.Y;
+			if (G.Y < minY) minY = G.Y;
+			if (H.Y < minY) minY = H.Y;
+
+			float minZ = A.Z;
+			if (B.Z < minZ) minZ = B.Z;
+			if (C.Z < minZ) minZ = C.Z;
+			if (D.Z < minZ) minZ = D.Z;
+			if (E.Z < minZ) minZ = E.Z;
+			if (F.Z < minZ) minZ = F.Z;
+			if (G.Z < minZ) minZ = G.Z;
+			if (H.Z < minZ) minZ = H.Z;
+
+			float maxX = A.X;
+			if (B.X > maxX) maxX = B.X;
+			if (C.X > maxX) maxX = C.X;
+			if (D.X > maxX) maxX = D.X;
+			if (E.X > maxX) maxX = E.X;
+			if (F.X > maxX) maxX = F.X;
+			if (G.X > maxX) maxX = G.X;
+			if (H.X > maxX) maxX = H.X;
+
+			float maxY = A.Y;
+			if (B.Y > maxY) maxY = B.Y;
+			if (C.Y > maxY) maxY = C.Y;
+			if (D.Y > maxY) maxY = D.Y;
+			if (E.Y > maxY) maxY = E.Y;
+			if (F.Y > maxY) maxY = F.Y;
+			if (G.Y > maxY) maxY = G.Y;
+			if (H.Y > maxY) maxY = H.Y;
+
+			float maxZ = A.Z;
+			if (B.Z > maxZ) maxZ = B.Z;
+			if (C.Z > maxZ) maxZ = C.Z;
+			if (D.Z > maxZ) maxZ = D.Z;
+			if (E.Z > maxZ) maxZ = E.Z;
+			if (F.Z > maxZ) maxZ = F.Z;
+			if (G.Z > maxZ) maxZ = G.Z;
+			if (H.Z > maxZ) maxZ = H.Z;
 
-			return new Box(
-				Math.Min(Math.Min(Math.Min(Math.Min(Math.Min(Math.Min(Math.Min(A, B), C), D), E), F), G), H),
```

Comment:
Pretty sure. `Min` and `Max` showed up as around 10% CPU each in the profiling. After the rewrite there is no trace of the `BoxTransform` method anywhere anymore.